sormas-backend/src/main/java/de/symeda/sormas/backend/common/AbstractCoreAdoService.java

```diff
@@ -195,8 +205,28 @@ public boolean inJurisdictionOrOwned(ADO entity) {
 		return fulfillsCondition(entity, this::inJurisdictionOrOwned);
 	}
 
+	public List<ADO> getEntitiesToBeProcessed(List<String> entityUuids, List<ProcessedEntity> processedEntities) {
+		List<String> failedUuids = processedEntities.stream()
+			.filter(
+				entity -> entity.getProcessedEntityStatus().equals(ProcessedEntityStatus.ACCESS_DENIED_FAILURE)
+					|| entity.getProcessedEntityStatus().equals(ProcessedEntityStatus.EXTERNAL_SURVEILLANCE_FAILURE))
+			.map(entity -> entity.getEntityUuid())
+			.collect(Collectors.toList());
+
+		List<ADO> entities = getByUuids(entityUuids);
+		return entities.stream().filter(entity -> !failedUuids.contains(entity.getUuid())).collect(Collectors.toList());
+	}
+
+	public List<ProcessedEntity> buildProcessedEntities(List<String> entityUuids, ProcessedEntityStatus processedEntityStatus) {
+		List<ProcessedEntity> processedEntities = new ArrayList<>();
+		entityUuids.forEach(entityUuid -> processedEntities.add(new ProcessedEntity(entityUuid, processedEntityStatus)));
```

Comment:
It can be solved with the one liner you've suggested but this method will be used in ExternalSurveillanceToolGatewayFacadeEjb too with different statuses too ( in order to avoid having a duplicate method buildProcessedEntitiesListByStatus(List<String> uuids, ProcessedEntityStatus processedEntityStatus)).

Till now I've created two methods buildProcessedEntities and a buildEntityByStatus and moved them to BaseAdoService since not only for the core entities I'll use them: 
![image](https://github.com/SORMAS-Foundation/SORMAS-Project/assets/47103965/7292475f-863c-44e0-ba4f-823cbe7d7c0a)

If we want to use them with methods from ProcessedEntity based on status, I've also created some of the methods ( I can create for all the statuses):
![image](https://github.com/SORMAS-Foundation/SORMAS-Project/assets/47103965/559c8e13-f646-41f2-8600-72fa33776780)

![image](https://github.com/SORMAS-Foundation/SORMAS-Project/assets/47103965/dbae2115-19a2-489c-8128-c760ce64f1ab)

If you think that would be worthy, the buildProcessedEntities method based on status can be used in other places too
where we are creating a new ProcessedEntity based on the status ( but there will be no major difference): 
![image](https://github.com/SORMAS-Foundation/SORMAS-Project/assets/47103965/f9ba48ee-cad0-4610-9b78-f460d869e392)

How would you prefer? :)

Reply:
I would say to create the methods just for the statuses that are used more often and it helps to write less and cleaner code